src/core/include/openvino/op/constant.hpp

```diff
@@ -511,6 +517,31 @@ class OPENVINO_API Constant : public Op {
         });
     }
 
+    template <element::Type_t Type, typename std::enable_if<Type == element::Type_t::string, bool>::type = true>
+    void cast_vector(std::vector<std::string>& output_vector, size_t num_elements) const {
+        auto output_size = std::min(num_elements, shape_size(m_shape));
+        output_vector.reserve(output_size);
+        const auto p = get_data_ptr<Type>();
+        std::copy(p, p + output_size, std::back_inserter(output_vector));
+    }
+
+    template <element::Type_t Type, typename std::enable_if<Type != element::Type_t::string, bool>::type = true>
+    void cast_vector(std::vector<std::string>& output_vector, size_t num_elements) const {
+        OPENVINO_ASSERT(false,
```

Comment:
`OPENVINO_THROW` looks to be designed for this purpose.

Reply:
Fixed.